statsmodels/doe/latin.py

```diff
@@ -0,0 +1,200 @@
+"""Latin Hypercube Sampling methods."""
+from __future__ import division
+import numpy as np
+from scipy.optimize import brute, fmin
+try:
+    from scipy.optimize import basinhopping
+    have_basinhopping = True
+except ImportError:
+    have_basinhopping = False
+from statsmodels.tools.sequences import discrepancy
+
+
+def orthogonal_latin_hypercube(dim, n_sample, bounds=None):
+    """Orthogonal array-based Latin hypercube sampling (OA-LHS).
+
+    On top of the constraints from the Latin Hypercube, an orthogonal array of
+    size n_sample is defined and only one point is allowed per subspace.
+
+    Parameters
+    ----------
+    dim : int
+        Dimension of the parameter space.
+    n_sample : int
+        Number of samples to generate in the parametr space.
+    bounds : tuple or array_like ([min, k_vars], [max, k_vars])
+        Desired range of transformed data. The transformation apply the bounds
+        on the sample and not the theoretical space, unit cube. Thus min and
+        max values of the sample will coincide with the bounds.
+
+    Returns
+    -------
+    sample : ndarray (n_samples, k_vars)
+        Latin hypercube Sampling.
+
+    References
+    ----------
+    [1] Art B. Owen, "Orthogonal arrays for computer experiments, integration
+    and visualization", Statistica Sinica, 1992.
+
+    """
+    sample = []
+    step = 1.0 / n_sample
+
+    for i in range(dim):
+        # Enforce a unique point per grid
+        temp = [np.random.uniform(low=j * step, high=(j + 1) * step)
+                for j in range(n_sample)]
```

Comment:
AFAICS, this can be vectorized, something like:

```
j = np.range(n_sample) * step
temp = j + np.random.uniform(low=0, high=step, size= n_sample)
```

Reply:
Oh great thanks. I actually kinda struggled about these lines!